pkg/reconciler/reconciler.go

```diff
@@ -755,6 +777,35 @@ func (r *Reconciler) doUpgrade(actionClient helmclient.ActionInterface, u *updat
 	return rel, nil
 }
 
+func (r *Reconciler) handlePending(actionClient helmclient.ActionInterface, rel *release.Release, u *updater.Updater, log logr.Logger) (ctrl.Result, error) {
```

Comment:
Could you add a new test case for a reconciliation with a `MarkFailedAfter` which tests the rollback from a failed to a successful release?
As these changes affect the core logic of our reconciliations it is important from my point of view.